PatternPal/PatternPal.Core/Recognizers/DecoratorRecognizer.cs

```diff
@@ -0,0 +1,442 @@
+    /// 5) Requirements for Client:
+    ///     a) has created an object of the type ConcreteComponent
+    ///     b) has created an object of the type ConcreteDecorator, to which it passes the ConcreteComponent
+    ///     c) has called the method of ConcreteDecorator
+    /// </summary>
+    public ICheck[] Checks()
+    {
+        ICheck[] result = new ICheck[5];
+
+        //Checks for requirement 1
+        MethodCheck componentMethod = ComponentMethod();
+        ICheck component = Component(componentMethod);
+
+        //Checks for requirement 2
+        ClassCheck concreteComponent = ConcreteComponent(component, componentMethod);
+
+        //Checks for requirement 3
+        FieldCheck baseDecoratorField = BaseDecoratorField(component);
+        MethodCheck baseDecoratorMethod = BaseDecoratorMethod(componentMethod, baseDecoratorField);
+        ClassCheck baseDecorator = BaseDecorator(component, baseDecoratorField, baseDecoratorMethod);
+
+        //Checks for requirement 4
+        MethodCheck concreteDecoratorExtraMethod = ConcreteDecoratorExtraMethod();
+        ClassCheck concreteDecorator = ConcreteDecorator(baseDecorator, concreteDecoratorExtraMethod, baseDecoratorMethod);
+
+        //Checks for requirement 5
+        ClassCheck client = Client(componentMethod, concreteDecorator, concreteComponent);
+
+        result[0] = component;
+        result[1] = concreteComponent;
+        result[2] = baseDecorator;
+        result[3] = concreteDecorator;
+        result[4] = client;
+
+        return result;
+    }
+
+    /// <summary>
+    /// Check for the <see cref="MethodCheck"/> of the Component of the Decorator pattern.
+    /// Used to check that there is a <see cref="IMethod"/> adhering to the requirements of the Component method.
+    /// </summary>
+    /// <remarks>
+    /// Checks part 1a of the requirements defined for a Decorator.
+    /// </remarks>
+    protected abstract MethodCheck ComponentMethod();
+
+    /// <summary>
+    /// Check for the of the Component of the Decorator pattern, which can be either a
+    /// <see cref="ClassCheck"/> or an <see cref="InterfaceCheck"/>.
+    /// Used to check that there is an <see cref="IEntity"/> adhering to the requirements of the Component.
+    /// </summary>
+    /// <remarks>
+    /// Checks part 1 of the requirements defined for a Decorator.
+    /// </remarks>
+    protected abstract ICheck Component(MethodCheck componentMethod);
+
+    /// <summary>
+    /// Check for the <see cref="RelationCheck"/> of both the ConcreteComponent and ConcreteDecorator.
+    /// Used to check that the found ConcreteDecorator and ConcreteComponent extend from Component.
+    /// </summary>
+    /// <remarks>
+    /// Checks part 2a and 3a of the requirements defined for a Decorator.
+    /// </remarks>
+    protected abstract RelationCheck ExtendsFrom(ICheck parent);
```

Comment:
```suggestion
    protected abstract RelationCheck Extends(ICheck parent);
```
`ExtendsFrom` sounds kind of weird, either you extend something, or you're extended from something.